### PR TITLE
JAXB implementation is a separate set of JAR files with split package issues

### PR DESCRIPTION
**Pull request by zhengjl**
@m0mus @lukasj @bravehorsie @Xiaojwu @HOHOWU

### CODE REVIEW

**Comment by bertramn** [comment]
Yes this library can do with some simplification of the package structure, well done!

File: jaxb-ri/runtime/impl/src/main/jdk9/module-info.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+/**
+ * The XML Binding (JAXB) RI modularization implementation.
+ *
+ * <p> This module is upgradeable.
+ *
+ * @uses javax.xml.bind.JAXBContextFactory
+ *
+ * @since 9
+ */
+module java.xml.bind.imp {
```
Comment (bravehorsie) [inline jaxb-ri/runtime/impl/src/main/jdk9/module-info.java]: Should impl be here?

File: jaxb-ri/runtime/impl/src/main/jdk9/module-info.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    requires java.xml.bind;
+    requires java.compiler;
+    requires java.desktop;
```
Comment (bravehorsie) [inline jaxb-ri/runtime/impl/src/main/jdk9/module-info.java]: What is desktop used for?

File: jaxb-ri/runtime/impl/src/main/jdk9/module-info.java [hunk 3]
```diff
@@ -58,9 +57,11 @@
     requires transitive java.activation;
     requires transitive java.xml;
 
-    exports com.sun.istack;
-    exports com.sun.istack.localization;
-    exports com.sun.istack.logging;
+    requires txw2;
```
Comment (bravehorsie) [inline jaxb-ri/runtime/impl/src/main/jdk9/module-info.java]: These requires (txw2, istack, etc) are copied by dependency plugin to module path. They will not be available for jaxb-ri.zip bundle, which has its classes repackaged.

**Comment by lukasj** [comment]
I'd perhaps go with "Automatic-Module-Name" header in manifest first instead of creating module-infos.[1] "Add-Exports" manifest header exists as well...

[1]: http://mail.openjdk.java.net/pipermail/jpms-spec-comments/2016-March/000024.html

**Comment by m0mus** [comment]
Looks good for me.

**Pull request merged**